pkg/controller/azvolumeattachment.go

```diff
@@ -124,11 +125,11 @@ func (r *reconcileAzVolumeAttachment) HandleAzVolumeAttachmentEvent(ctx context.
 		// if the role in status and spec are different, it is an update event where replica should be turned into a primary
 	} else if azVolumeAttachment.Spec.RequestedRole != azVolumeAttachment.Status.Role {
 		klog.Infof("Promoting AzVolumeAttachment (%s) from replica to primary", azVolumeAttachment.Name)
-		if err := r.UpdateStatus(ctx, azVolumeAttachment.Name); err != nil {
+		if err := r.UpdateStatus(ctx, azVolumeAttachment.Name, nil); err != nil {
```

Comment:
If we pass `nil` status, `azVolumeAttachment.Status.ResponseObject` will be overwritten with `nil`.

Since this part of the function involves promotion of replica AzVolumeAttachment to primary attachment, I don't think its `ResponseObject` should be overwritten with `nil` value unless this is intended for certain crd provisioner logic.